### ISSUE TITLE
`wkhtmltopdf http://google.com google.pdf` gives HostNotFoundError 

### ISSUE THREAD

**Issue by promilo** [wkhtmltopdf/wkhtmltopdf#4278]
**wkhtmltopdf version(s) affected**: wkhtmltox-0.12.4_linux-generic-amd64.tar.xz

**OS information**

alpine 3.9

**Description**

```
wkhtmltopdf http://google.com google.pdf
Loading pages (1/6)
Error: Failed loading page http:// (sometimes it will work just to ignore this error with --load-error-handling ignore)
Exit with code 1 due to network error: HostNotFoundError
``` 
**How to reproduce**

```
wkhtmltopdf http://google.com google.pdf
``` 

**Expected behavior**

should generate pdf

**Possible Solution**

**Comment by mvincent514** [comment]
Hi,
Have you tried :
wkhtmltopdf https://google.com google.pdf
Maybe it's intented by google not to serve plain http with no ssl.
Can't try it myself since i don't use wkhtmltopdf yet.

**Comment by stevenwilliamolthoff** [comment]
Did you find a solution?

**Comment by stevenwilliamolthoff** [comment]
I found that I had to use a version of wkhtmltopdf that is patched for Alpine. github.com/madnight/docker-alpine-wkhtmltopdf

**Comment by marksansome** [comment]
I had the same issue. Was able to fix it using a patched version of `wkhtmltopdf` for Alpine here https://github.com/RoseRocket/docker-alpine-wkhtmltopdf-patched-qt. Very similar solution to @stevenwilliamolthoff

**Comment by PhilterPaper** [comment]
If you use the right URL _in the first place_ (**https://...**), does it work? I've seen a 30x status come back to redirect to https or add/delete www., that wkHTMLtoPDF couldn't handle.

**Comment by marksansome** [comment]
> If you use the right URL _in the first place_ (**https://...**), does it work? I've seen a 30x status come back to redirect to https or add/delete [www](http://www)., that wkHTMLtoPDF couldn't handle.

Yes, even when using `https://` and/or `www.` we see the same response of `HostNotFoundError`

**Comment by PhilterPaper** [comment]
So `wkHTMLtoPDF https://www.google.com google.pdf` fails? It works fine for me (0.12.5 with patched qt, on Windows 10). Even `http://google.com google.pdf` works. Are you sure that google.com isn't being blocked by your local network?

**Comment by marksansome** [comment]
@PhilterPaper please re-read the original issue. As described the problem is occurring with `wkhtmltopdf 0.12.4` on Alpine 3.9. The solution we found to this was to manually patch wkhtmltopdf 0.12.5 for Alpine as seen here https://github.com/wkhtmltopdf/wkhtmltopdf/issues/4278#issuecomment-504151541

**Comment by mjarkk** [comment]
Is there a reason why we need to use a patched version to get this working on alpine linux?  
This is quite annoying to debug because it looks like it's a problem made by the user not the library.

**Comment by promilo** [comment]
Manually patching it to 0.12.5 fixed the issue! thanks a lot @marksansome and @stevenwilliamolthoff
